docs/source/ProgrammingGuide/HierarchicalParallelism.md

```diff
@@ -73,12 +74,13 @@ TeamPolicy<ExecutionSpace> policy (league_size, Kokkos::AUTO() );
 // Launch a kernel
 parallel_for (policy, KOKKOS_LAMBDA (member_type team_member) {
     // Calculate a global thread id
-     int k = team_member.league_rank () * team_member.team_size () +
+    int k = team_member.league_rank () * team_member.team_size () +
             team_member.team_rank ();
     // Calculate the sum of the global thread ids of this team
-     int team_sum = team_member.reduce (k);
-     // Atomically add the value to a global value
-     a() += team_sum;
+    int team_sum = k;
+    team_member.team_reduce(Sum<int>(team_sum));
```

Comment:
```suggestion
    team_member.team_reduce(Sum<int, typename ExecutionSpace::memory_space>(team_sum));
```
since the referenced value is not in `HostSpace`. We would only check the memory space if accessing the respective `View`, though.